homeassistant/helpers/restore_state.py

```diff
@@ -2,97 +2,145 @@
 import asyncio
 import logging
 from datetime import timedelta
+from typing import Any, Dict, List, Optional  # noqa  pylint_disable=unused-import
 
-import async_timeout
+from homeassistant.core import HomeAssistant, callback, State, CoreState
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.json import JSONEncoder
+from homeassistant.helpers.storage import Store  # noqa  pylint_disable=unused-import
 
-from homeassistant.core import HomeAssistant, CoreState, callback
-from homeassistant.const import EVENT_HOMEASSISTANT_START
-from homeassistant.loader import bind_hass
-from homeassistant.components.history import get_states, last_recorder_run
-from homeassistant.components.recorder import (
-    wait_connection_ready, DOMAIN as _RECORDER)
-import homeassistant.util.dt as dt_util
+DATA_RESTORE_STATE_TASK = 'restore_state_task'
 
-RECORDER_TIMEOUT = 10
-DATA_RESTORE_CACHE = 'restore_state_cache'
-_LOCK = 'restore_lock'
 _LOGGER = logging.getLogger(__name__)
 
+STORAGE_KEY = 'core.restore_state'
+STORAGE_VERSION = 1
+
+# How long between periodically saving the current states to disk
+STATE_DUMP_INTERVAL = timedelta(minutes=15)
+
+
+class RestoreStateData():
+    """Helper class for managing the helper saved data."""
+
+    @staticmethod
```

Comment:
Shouldn't this be a `classmethod` since we create and return an instance of this class. Thats a common use case for `classmethod`.